### ISSUE TITLE
Does not compile on FreeBSD anymore

### ISSUE THREAD

**Issue by Luflosi** [kovidgoyal/kitty#2187]
2cee3a88094d2421e74738e59de97abd3302a20f broke compilation on FreeBSD:
```
freebsd% gmake                                                
python3 setup.py 
CC: gcc (9, 2)
<stdin>: In function 'main':
<stdin>:5:24: error: '__NR_memfd_create' undeclared (first use in this function)
<stdin>:5:24: note: each undeclared identifier is reported only once for each function it appears in
[5/84] Compiling kitty/fonts.c ... done
Compiling [wayland] glfw/wl_window.c ...
gcc -MMD -DNDEBUG -D_GLFW_WAYLAND -D_GLFW_BUILD_DLL -Wextra -Wfloat-conversion -Wno-missing-field-initializers -Wall -Wstrict-prototypes -D_POSIX_C_SOURCE=200809L -std=c11 -pedantic-errors -Werror -O3 -fwrapv -fstack-protector-strong -pipe -march=native -fvisibility=hidden -fPIC -flto -fpic -pthread -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include/dbus-1.0 -I/usr/local/lib/dbus-1.0/include -c glfw/wl_window.c -o build/glfw-wayland-wl_window.c.o
In file included from glfw/wl_window.c:33:
glfw/memfd.h: In function 'createTmpfileCloexec':
glfw/memfd.h:47:10: error: implicit declaration of function 'mkostemp'; did you mean 'mkstemp'? [-Wimplicit-function-declaration]
   47 |     fd = mkostemp(tmpname, O_CLOEXEC);
      |          ^~~~~~~~
      |          mkstemp
glfw/wl_window.c: In function 'read_primary_selection_offer':
glfw/wl_window.c:1549:9: error: implicit declaration of function 'pipe2'; did you mean 'pipe'? [-Wimplicit-function-declaration]
 1549 |     if (pipe2(pipefd, O_CLOEXEC) != 0) return NULL;
      |         ^~~~~
      |         pipe
gmake: *** [Makefile:9: all] Error 1
```
Is there a good way to fix this? Perhaps setting `_POSIX_C_SOURCE` to a different value on FreeBSD or using different functions instead of the undeclared ones?

**Comment by Luflosi** [comment]
The commit causes similar error messages on OpenBSD.

**Comment by kovidgoyal** [comment]
I have no idea, I dont touch BSDs. Presumably they decided to not include functions that require _GNU_SOURCE if _POSIX_C_SOURCE is defined. Your best bet to fix all these issues is probably to refactor monotonic.h. Have it only expose monotonic() and the conversion functions and monotonic_t. The move the implementation of monotonic_() which requires clock_gettime() and therefore _POSIX_C_SOURCE into its own .c files one in glfw and one in kitty. And in that .c file include time.h above monotonic.h and define _POSIX_C_SOURCE there and not as a compiler define.

**Comment by Luflosi** [comment]
Awesome, you fixed it already 🎉.
Do you think that this also fixes #2172?

**Comment by kovidgoyal** [comment]
No idea, it might.
